lib/core/api.js

```diff
@@ -252,12 +256,22 @@ function postBlockToPlatform(data, groupId, blockIndex, callback) {
 
   api.log('postBlockToPlatform #' + blockIndex + ': using id ', groupId);
 
-  tidepool.uploadDeviceDataForUser(groupId, data, function(err, result) {
-    if (err) {
-      return sad(err, err);
-    }
-    return happy(result);
-  });
+  if(dataservices) {
+    tidepool.uploadDataServicesDataForUser(groupId, data, function(err, result) {
```

Comment:
The only real difference is whether we're calling `tidepool.uploadDataServiceDataForUser` or `tidepool.uploadDeviceDataForUser`. I've refactored it to remove the confusing if statements based on the boolean variable, and pass in the `uploadType` instead, that is then resolved to the correct function using `getUploadFunction()`. Does that make the code a bit clearer?